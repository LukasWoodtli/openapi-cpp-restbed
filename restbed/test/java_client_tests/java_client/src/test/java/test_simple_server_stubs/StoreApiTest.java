package test_simple_server_stubs;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.openapi.example.api.StoreApi;
import org.openapi.example.model.Order;

import static helper.ApiClientFactories.setUpStoreApi;
import static helper.TestingHelper.approveResponseAsJson;
import static org.junit.jupiter.api.Assertions.*;

class StoreApiTest {

    StoreApi apiInstance;

    @BeforeEach
    void setUp() {
        apiInstance = setUpStoreApi(1235);
    }

    @Test
    void deleteOrder() {
        apiInstance.deleteOrder("myOrderId").block();
    }

    @Test
    void getInventory() {
        var inventory = apiInstance.getInventory().block();
        approveResponseAsJson(inventory);
    }

    @Test
    void getOrderById() {
        var order = apiInstance.getOrderById(123L).block();
        approveResponseAsJson(order);
    }

    @Test
    void placeOrder() {
        var order = new Order();
        order.setId(9876L);
        order.setStatus(Order.StatusEnum.PLACED);
        var response = apiInstance.placeOrder(order).block();
        approveResponseAsJson(response);
    }
}