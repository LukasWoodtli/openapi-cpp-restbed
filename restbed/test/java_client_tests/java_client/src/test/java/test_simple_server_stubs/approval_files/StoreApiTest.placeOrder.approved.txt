{
  "id" : 9876,
  "petId" : 0,
  "quantity" : 0,
  "shipDate" : {
    "dayOfMonth" : 12,
    "dayOfWeek" : "WEDNESDAY",
    "dayOfYear" : 12,
    "hour" : 23,
    "minute" : 24,
    "month" : "JANUARY",
    "monthValue" : 1,
    "nano" : 0,
    "offset" : {
      "id" : "Z",
      "rules" : {
        "fixedOffset" : true,
        "transitionRules" : [ ],
        "transitions" : [ ]
      },
      "totalSeconds" : 0
    },
    "second" : 2,
    "year" : 2011
  },
  "status" : "placed",
  "complete" : false
}